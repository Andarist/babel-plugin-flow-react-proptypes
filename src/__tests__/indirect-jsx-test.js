var babel = require('babel-core');
var content = `
type Props = {
  name: string,
}

var C = (props: Props) => {
  var el = null;
  if (true) {
    el = <div />;
  }
  return el;
};
`;

it('indirect-jsx', () => {
  var res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
