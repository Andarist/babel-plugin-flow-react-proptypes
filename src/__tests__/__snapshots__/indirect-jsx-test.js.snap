// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indirect-jsx 1`] = `
"\\"use strict\\";

var _propTypes = require(\\"prop-types\\");

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

C.propTypes = {
  name: _propTypes2.default.string.isRequired
};


var C = function C(props) {
  var el = null;
  if (true) {
    el = React.createElement(\\"div\\", null);
  }
  return el;
};"
`;
